src/ert/storage/local_ensemble.py

```diff
@@ -574,13 +662,122 @@ def load_parameters(
             Loaded xarray Dataset with parameters.
         """
 
-        return self._load_dataset(group, realizations)
+        # UGLY AF, would be better to narrow in realizations
+        # to be either None of a tuple of ints
+        drop_reals_dim = False
+        selected_realizations: Union[None, int, List[int]] = []
+        if realizations is None:
+            selected_realizations = None
+        elif type(realizations) is int:
+            assert type(realizations) is int
+            drop_reals_dim = True
+            selected_realizations = realizations
+        elif type(realizations) is np.ndarray:
+            selected_realizations = realizations.tolist()
+        elif isinstance(realizations, tuple):
+            selected_realizations = list(realizations)
+        elif isinstance(realizations, list):
+            selected_realizations = realizations
+        else:
+            raise ValueError(f"Invalid type for realizations: {type(realizations)}")
+
+        try:
+            ds = self.open_unified_parameter_dataset(group)
+            if selected_realizations is None:
+                return ds
+
+            return ds.sel(realizations=selected_realizations, drop=drop_reals_dim)
```

Comment:
Will this require `list`? Otherwise you could use `Iterable` in the typing, and get rid of a lot of the checking above.

Reply:
Iterable almost works. It seems `range` works, but tuple doesnt, and tuple is also `Iterable`